packages/react-native-sdk/src/components/Livestream/LivestreamTopView/DurationBadge.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { useTheme } from '../../../contexts';
+import { ShieldBadge } from '../../../icons';
+import { useCall } from '@stream-io/video-react-bindings';
+import { CallSessionResponse, StreamCallEvent } from '@stream-io/video-client';
+
+/**
+ * Props for the HostDurationBadge component.
+ */
+export type DurationBadgeProps = {
+  mode?: 'host' | 'viewer';
+};
+
+/**
+ * The HostDurationBadge component displays the duration while the live stream is active.
+ */
+export const DurationBadge = ({ mode }: DurationBadgeProps) => {
+  const [duration, setDuration] = useState(0);
+  const call = useCall();
+  const {
+    theme: {
+      colors,
+      variants: { iconSizes },
+      durationBadge,
+    },
+  } = useTheme();
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timer;
+    const handleLiveStarted = () => {
+      intervalId = setInterval(() => {
+        setDuration((d) => d + 1);
+      }, 1000);
+    };
+
+    const handleLiveEnded = (event: StreamCallEvent) => {
+      const callDetails = (
+        event as StreamCallEvent & {
+          call: { session: CallSessionResponse };
+        }
+      ).call.session;
+      if (callDetails?.live_ended_at !== null) {
+        clearInterval(intervalId);
+      }
+    };
+
+    if (mode === 'host') {
+      call?.on('call.live_started', handleLiveStarted);
+      call?.on('call.updated', handleLiveEnded);
+    } else {
+      handleLiveStarted();
+    }
+
+    return () => {
+      if (mode === 'host') {
+        call?.off('call.live_started', handleLiveStarted);
+        call?.off('call.updated', handleLiveEnded);
```

Comment:
Instead of `off`, use the unsubscribe functions returned by `on` function. We ran into some issue with current approach on chat in the past.

Reply:
Done @vishalnarkhede